File: 434.py
class Solution:
    def countSegments(self, s: str) -> int:
        # time: O(n), space: O(1)
        cnt = 0
        isSegment = False
        for c in s:
            if c != " ":
                isSegment = True
            else:
                if isSegment:
                    cnt += 1
                isSegment = False
        return cnt + 1 if isSegment else cnt

File: 434.java
class Solution {
    public int countSegments(String s) {
        // time: O(n), space: O(1)
        int cnt = 0;
        for (int i = 0; i < s.length(); i++) {
            if ((i == 0 || s.charAt(i - 1) == ' ') && s.charAt(i) != ' ') {
                cnt++;
            }
        }
        return cnt;
    }
}